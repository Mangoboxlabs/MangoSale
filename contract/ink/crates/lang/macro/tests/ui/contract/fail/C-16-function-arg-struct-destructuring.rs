use ink_lang as ink;

#[ink::contract]
mod noop {

    struct Args {
        foo: (),
    }

    #[ink(storage)]
    pub struct Noop {}

    impl Noop {
        #[ink(constructor)]
        pub fn new() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn noop(&self, Foo { foo }: Foo) {}
    }
}

fn main() {}
