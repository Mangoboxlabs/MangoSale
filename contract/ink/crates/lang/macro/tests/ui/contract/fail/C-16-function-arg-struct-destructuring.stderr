error: ink! message arguments must have an identifier
  --> $DIR/C-16-function-arg-struct-destructuring.rs:20:28
   |
20 |         pub fn noop(&self, Foo { foo }: Foo) {}
   |                            ^^^^^^^^^^^^^^^^
